dragonboard/io.py

```diff
@@ -4,179 +4,94 @@
-    '''
-    d = np.fromfile(f, '>i2', num_gains * num_channels * roi)
-
-    N = num_gains * num_channels * roi
-    roi_dtype = '{}>i2'.format(roi)
-    array = np.empty(
-        num_channels, dtype=[('low', roi_dtype), ('high', roi_dtype)]
-    )
-    data_odd = d[N / 2:]
-    data_even = d[:N / 2]
-    for channel in range(0, num_channels, 2):
-        array['high'][channel] = data_even[channel::8]
-        array['low'][channel] = data_even[channel + 1::8]
-        array['high'][channel + 1] = data_odd[channel::8]
-        array['low'][channel + 1] = data_odd[channel + 1::8]
-
-    return array
-
-
-def guess_event_size(f):
-    ''' try to find out the event size for this file.
-
-    Each even header contains a flag.
-    The distance between two flags is just the event size.
-    '''
-    current_position = f.tell()
-    f.seek(0)
-
-    max_event_size = get_event_size(roi=max_roi)
-    # I don't believe myself, so I add 50% here
-    chunk_size = int(max_event_size * 1.5)
-
-    chunk = f.read(chunk_size)
-
-    # the flag should be found two times in the chunk:
-    #  1.) in the very first 48 bytes as part of the first event header
-    #  2.) somewhere later, as part of the second header.
-    # the distance is simply the event size:
-    #
-    # Note! At first i though the flag is always this: flag = b'\xf0\x02' * 8
-    # But then I found files, where this is not true,
-    # Now I make another assumption about the flag.
-    # I assume: The flag is the the bytestring from address 16 to 32 of the
-    # file:
-
-    flag = chunk[16:32]
-    first_flag = chunk.find(flag)
-    second_flag = chunk.find(flag, first_flag + 1)
-
-    event_size = second_flag - first_flag
-    f.seek(current_position)
-
-    return event_size
+adc_word_size = 2
 
 
 def read(path, max_events=None):
     ''' return list of Events in file path '''
-    with open(path, 'rb') as f:
-        return list(EventGenerator(f, max_events=None))
+    return list(EventGenerator(path, max_events=None))
 
+class AbstractEventGenerator(object):
+    header_size = None
 
-class EventGenerator(object):
-    def __init__(self, file_descriptor, max_events=None):
-        self.file_descriptor = file_descriptor
-        self.event_size = guess_event_size(file_descriptor)
+    def __init__(self, path, max_events=None):
+        self.path = os.path.realpath(path)
+        self.file_descriptor = open(self.path, "rb")
+        self.event_size = self.guess_event_size()
+        self.roi = self.get_roi()
         self.max_events = max_events
+        self.get_length()
+
+    def __str__(self):
+        return(
+            "{name}(\n"
+            "path={S.path!r}, \n"
+            "max_events={S.max_events})\n"
+            "roi ......: {S.roi}\n"
+            "event_size: {S.event_size} byte\n"
+            "#events ..: {S._length}"
+        ).format(
+            name=self.__class__.__name__,
+            S=self
+        )
+
+    def get_length(self):
```

Comment:
Every other get_method returns something, this just sets an instance member. I think this shoudl be a private method like `_calc_num_events` and return it.